dart/lib/vm_service_lib.dart

```diff
@@ -132,7 +132,664 @@ Map<String, Function> _typeFactories = {
   'HeapSpace': HeapSpace.parse,
 };
 
-class VmService {
+abstract class VmServiceInterface {
+  /// Returns the stream for a given stream id.
+  ///
+  /// This is not a part of the spec, but is needed for both the client and
+  /// server to get access to the real event streams.
+  Stream<Event> onEvent(String streamId);
+
+  /// The `addBreakpoint` RPC is used to add a breakpoint at a specific line of
+  /// some script.
+  ///
+  /// The `scriptId` parameter is used to specify the target script.
+  ///
+  /// The `line` parameter is used to specify the target line for the
+  /// breakpoint. If there are multiple possible breakpoints on the target line,
+  /// then the VM will place the breakpoint at the location which would execute
+  /// soonest. If it is not possible to set a breakpoint at the target line, the
+  /// breakpoint will be added at the next possible breakpoint location within
+  /// the same function.
+  ///
+  /// The `column` parameter may be optionally specified. This is useful for
+  /// targeting a specific breakpoint on a line with multiple possible
+  /// breakpoints.
+  ///
+  /// If no breakpoint is possible at that line, the `102` (Cannot add
+  /// breakpoint) error code is returned.
+  ///
+  /// Note that breakpoints are added and removed on a per-isolate basis.
+  ///
+  /// See [Breakpoint].
+  Future<Breakpoint> addBreakpoint(String isolateId, String scriptId, int line,
+      {int column});
```

Comment:
Not for this PR, but I wonder if we should have some simple heuristic for the number of args, or the string length of the combined args, and add a trailing comma to the arg list.

Reply:
Well, I personally really dislike the style of trailing commas in function definitions and invocations :D.

My personal preference aside though I think its better to be consistent about it as opposed to choosing to add the trailing comma sometimes and not others, so I would prefer always providing it or never.